test/integration/protocol_integration_test.cc

```diff
@@ -2997,6 +2997,180 @@ TEST_P(DownstreamProtocolIntegrationTest, LocalReplyWithMetadata) {
   ASSERT_EQ("200", response->headers().getStatusValue());
 }
 
+TEST_P(DownstreamProtocolIntegrationTest, LocalReplyFromDecodeMetadata) {
```

Comment:
can we move this to filter_integration_test and make sure that for both upstream and downstream if we have
StopAllFilter MetadataLocalReplyFilter ReleaseAssertIfSeesHeadersFilter RouterFilter
that all goes well?
I have some concerns about the filter manager impl and want to make sure tests cover things

Reply:
Added test that does MetadataLocalReplyFilter -> ReleaseAssertIfSeesHeadersFilter, but am not sure what the purpose of the StopAllFilter is.

Reply:
is this now redundant with the filter integration tst?  should we remove?